### PR TITLE
DB-9803 Implement GET_NEAREST_TRANSACTION sql function.

### PR DESCRIPTION
**Pull request by hatyo**

### CODE REVIEW

File: splice_machine/src/test/java/com/splicemachine/derby/transactions/GetNearestTransactionIT.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        Assert.assertTrue(rs.wasNull());
+        Assert.assertFalse(rs.next());
+    }
+
+    @Test
+    public void GetNearestTransactionWorksCorrectlyWithAutoCommit() throws Exception {
+        try {
+            long result1 = resultOf("VALUES get_nearest_Transaction(CURRENT_TIMESTAMP)", connACOff);
+            long result2 = resultOf("VALUES get_nearest_Transaction(CURRENT_TIMESTAMP)", connACOff);
+            Assert.assertEquals(result1, result2);
```
Comment (yxia92) [inline splice_machine/src/test/java/com/splicemachine/derby/transactions/GetNearestTransactionIT.java:81]: If there are other ITs running concurrently at the time of this test, and there is another write txn goes in between the two calls of `values get_nearest_transaction(current_timestamp)`, would we guarantee to get result1=result2?

**Review by yxia92 — CHANGES_REQUESTED** [review]

**Review by ascend1 — APPROVED** [review]
Other than Yi's non-reserved keyword comment, I think the change is good.

File: splice_machine/src/test/java/com/splicemachine/derby/transactions/GetNearestTransactionIT.java (hunk 1 above)
Comment (hatyo) [reply splice_machine/src/test/java/com/splicemachine/derby/transactions/GetNearestTransactionIT.java:81]: You're right.

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/sqlgrammar.jj [hunk 2]
```diff
@@ -17796,6 +17818,7 @@ reservedKeyword() :
 |    tok = <FUNCTION>
 |    tok = <GET>
 |    tok = <GET_CURRENT_CONNECTION>
+|    tok = <GET_NEAREST_TRANSACTION>
```
[the remark this answers is not in the document]
Comment (hatyo) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/sqlgrammar.jj:17821]: done.

**Comment by hatyo** [comment]
Thanks Yi and Zhen for the comments, could you please have another look?

**Comment by hatyo** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1559/

**Comment by hatyo** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Review by yxia92 — APPROVED** [review]
+1

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1561/

**Review by carolp-503 — APPROVED** [review]
+1

**Review by jyuanca — APPROVED** [review]

**Pull request merged**